src/app/components/all-products-components/all-products-content/all-products-content.component.ts

```diff
@@ -9,13 +11,57 @@ import {ProductCard} from "../../../interfaces/product-card";
   encapsulation: ViewEncapsulation.None
 })
 export class AllProductsContentComponent implements OnInit {
+
   @Input() products: ProductCard[] = [];
-  @Input() productsQuantity?:number;
-  p: number = 3;
+  @Input() productsQuantity: number;
+  public p: number = 3;
+  public itemsPerPage: number = 5;
+  public farmValue: string[] = [];
+  public filteredItems: ProductCard[] = [];
+
 
-  constructor() { }
+  constructor(private filtersService: FiltersService) {}
 
   ngOnInit(): void {
+    this.sendItemsPerPage();
+  }
+
+
+  sendItemsPerPage(): void {
+    this.filtersService.itemsPerPage.next(this.itemsPerPage);
   }
 
+  getFarmValue(event: MatCheckboxChange): void {
+    this.products = this.sortByFarm(event);
+  }
+
+  sortByFarm(event: MatCheckboxChange): any {
+    if (event.checked) {
+      this.farmValue.push(event.source.value);
+      this.filteredItems = this.filtersService.products.value.filter((item: ProductCard) => {
+        if (this.farmValue.includes(item.farm)) {
+          return true;
+        }
+      });
+      this.p = 1;
+      return this.filteredItems;
+    } else if (!event.checked) {
+      this.farmValue = this.farmValue.filter((item: string) => {
+        if (item === event.source.value) {
+          return false;
+        } else {
+          return true;
+        }
+      });
+      this.filteredItems = this.filtersService.products.value.filter((item: ProductCard) => {
+        if (this.farmValue.includes(item.farm)) {
+          return true;
+        } else if (this.farmValue.length === 0) {
+          return this.filteredItems;
+        }
+      });
+      return this.filteredItems;
```

Comment:
Separate this huge method into a few small ones

Reply:
+